src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs/Tabs.js

```diff
@@ -1,56 +1,398 @@
+            if (this.tabWidths.get(key) !== width) {
+                this.tabWidths.set(key, width);
+            }
+        });
+    };
+
+    setDimensions = () => {
+        this.setWrapperWidth();
+        this.setTabsWrapperWidth();
+        this.setTabsWidth();
+        this.updateTabWidths();
+    };
+
+    @action handleTabRefChange = (index: number, ref: ?ElementRef<'li'>) => {
+        if (this.tabRefs.get(index) !== ref) {
+            this.tabRefs.set(index, ref);
+        }
+    };
+
+    @action handleDropdownToggle = () => {
+        this.dropdownOpen = !this.dropdownOpen;
+    };
+
+    @action handleDropdownClose = () => {
+        this.dropdownOpen = false;
+    };
+
+    changeTab = (selectedTabIndex: ?number) => {
+        if (selectedTabIndex !== undefined && selectedTabIndex !== null) {
+            this.props.onSelect(selectedTabIndex);
+        }
+    };
+
+    handleTabClick = (selectedTabIndex: ?number) => {
+        this.changeTab(selectedTabIndex);
+    };
+
+    @action handleCollapsedTabClick = (selectedTabIndex: number) => {
+        this.dropdownOpen = false;
+        this.lastSelectedIndex = selectedTabIndex;
+
+        this.changeTab(selectedTabIndex);
+    };
+
     isSelected(tabIndex: number) {
         return tabIndex === this.props.selectedIndex;
     }
 
-    createTabItems(tabs: ChildrenArray<Element<typeof Tab> | false>) {
-        return React.Children.map(tabs, (tab, index) => {
+    get childIndices(): number[] {
+        const {children} = this.props;
+
+        return React.Children.map(children, (child, index) => index);
+    }
+
+    @computed get visibleTabIndices(): number[] {
+        if (this.tabsWidth <= this.wrapperWidth) {
+            return this.childIndices;
+        }
+
+        const {selectedIndex} = this.props;
+
+        let visibleWidth = 0;
+        let visibleTabIndices: number[] = [];
+
+        const childIndices = this.childIndices;
+        childIndices.sort((a, b) => {
+            if (a === selectedIndex) {
+                return -1;
+            }
+
+            if (b === selectedIndex) {
+                return 1;
+            }
+
+            if (a === this.lastSelectedIndex) {
+                return -1;
+            }
+
+            if (b === this.lastSelectedIndex) {
+                return 1;
+            }
+
+            return a - b;
+        });
+
+        for (const index of childIndices) {
+            const nextWidth = this.tabWidths.get(index);
+
+            if (undefined === nextWidth) {
+                if (visibleTabIndices.length > 0) {
+                    break;
+                }
+
+                return this.childIndices;
+            }
+
+            if (visibleWidth + nextWidth > this.tabsWrapperWidth) {
+                break;
+            }
+
+            visibleWidth += nextWidth;
+            visibleTabIndices = [...visibleTabIndices, index];
+        }
+
+        visibleTabIndices.sort((a, b) => a - b);
```

Comment:
What is this `sort` call at the very end doing?

Reply:
previously all tabs are sorted that the current selected index is at first position and the last selected index is at second position. then we calculate how many tabs can be displayed and then all the visible tabs are sorted in their natural order again. this ensures, that the selected tab and the last selected tab are always shown.

Reply:
Ah, I understand... I would say that this is no obvious, and needs some comment. I would probably add at the top where you move it to the beginning (another `sort` call where you set the value to `-1`), that we move the currently selected tab to the very beginning in order to make sure it is included in the visible tab list, and add a comment here that we have to sort again, because the currently selected item has been put at the very front, which we need to fix now.